### ISSUE TITLE
Purchases.getPurchaserInfo() doesn't return active subscriptions

### ISSUE THREAD

**Issue by ivangr1** [RevenueCat/react-native-purchases#115]
**TLDR**
``Purchases.getPurchaserInfo()`` says there are no active subscriptions if I leave the component where I purchased it or reopen the app

-----
I followed you docs and implemented Purchases SDK in my app and everything works fine except when I try to find if the user is subscribed.
I have a screen with a button that opens the paywall modal if the user is not subscribed and after he makes the purchase it closes the modal and the same button now does some other thing, pretty basic stuff.
This is screen component, only the relevant code:

```
_checkSubscriber = async ()=> {
    try {
      const purchaserInfo = await Purchases.getPurchaserInfo();
      console.log(purchaserInfo);
      if (typeof purchaserInfo.entitlements.active.subscriber !== 'undefined') {
        // some action only for subscribers
      } else {
        this.setState({paywallVisible: true});
      }
    } catch (e) {
      console.log(e);
    }
  };
render() {
<Button
     onPress={() => this._checkSubscriber()}
 />
}
```
and this is the paywall component:
```
async componentDidMount() {
    try {
      const offerings = await Purchases.getOfferings();
      if (
        offerings.current !== null &&
        offerings.current.availablePackages.length !== 0
      ) {
        this.setState({
          packages: offerings.current.availablePackages,
        });
      }
    } catch (e) {
      console.log(e);
    }
  } 
 _purchase = async pckg => {
    try {
      const {purchaserInfo} = await Purchases.purchasePackage(pckg);
      if (typeof purchaserInfo.entitlements.active.subscriber !== 'undefined') {
        this.props.closePaywall();
        });
      }
    } catch (e) {
      if (!e.userCancelled) {
        console.log(e);
      }
    }
  };
render() {
<Button
    onPress={() => this._purchase(packages[selectedPackage])}
 />
}
```

It says in your docs that the latest information can always be retrieved by calling ``Purchases.getPurchaserInfo()``, and everything works fine immediately after purchase, but when I leave that screen or reopen the app ``getPurchaserInfo()`` says there are no active subscriptions.
If I restore the purchase it works again but only if I don't leave the app, and sometimes when I change the screens, but not always.
I have also tried using PurchaserInfoUpdateListener inside componentDidMount() and then saving the subscription status inside state and checking it later, but then it fails even more frequently.

I've tried sandbox and real acounts on a simulator and on real device and it is the same for Android and iOS.

I am using Purchases version 3.0.6 with React Native 0.61.5

Can you help me, am I doing something wrong?

**Comment by vegaro** [comment]
Hi @ivangr1, sorry we forgot to reply here.

Please take into account the subscriptions are short lived in sandbox environments and they don't match the production durations. Subscriptions in sandbox normally last around 5 minutes.

How are you initializing your SDK? Are you using anonymous users or are you specifying a user when initializing?

**Comment by aboedo** [comment]
closing out this ticket due to inactivity. @ivangr1 feel free to reopen if needed.
